eth2/lmd_ghost/src/reduced_tree.rs

```diff
@@ -276,55 +305,62 @@ where
         Ok(weight)
     }
 
+    /// Removes the vote from `validator_index` from the reduced tree.
+    ///
+    /// If the validator had a vote in the tree, the removal of that vote may cause a node to
+    /// become redundant and removed from the reduced tree.
     fn remove_latest_message(&mut self, validator_index: usize) -> Result<()> {
         if self.latest_votes.get(validator_index).is_some() {
             // Unwrap is safe as prior `if` statements ensures the result is `Some`.
-            let vote = self.latest_votes.get(validator_index).unwrap();
+            //
+            // We do this instead of `if let` to satisfy the borrow checker.
```

Comment:
You could try `if let Some(vote) = self.latest_votes.get(validator_index).copied()`, which should release the borrow of `self.latest_votes` allowing you to mutate `self` later.